### PR TITLE
removes modelDefinition.__name from models hook

### PR DESCRIPTION
**Pull request by bschnelle**
as far as I can tell this attribute isn't being used anywhere and it is causing vogels.js to throw an error when creating models

### CODE REVIEW

**Comment by cludden** [comment]
there needs to be some way for the adapter to have access to the final model name. Both sequelize and mongoose require a name for defining a model, and it seems dumb to require the user to define the filename in every model definition. The only other alternative I can think of is to add the name as an argument to the registerModel hook:

``` javascript
registerModel: function(connection, definition, name, callback) {

}
```

to make this a nonbreaking change, we could make the name argument optional. thoughts?

**Comment by bschnelle** [comment]
do you just delete definition.__name in your adapter prior to passing it along to mongoose/sequelize?

**Comment by cludden** [comment]
i haven't landed on the final approach, but im thinking something like

``` javascript
// in /app/models/blog/post.js
module.exports = function(connection, Schema, name) {
    var schema = new Schema({
        title: String,
        body: String,
        date: Date
    });

    return connection.model(name, schema);
}

// in the adapter
{
    registerModel: function(connection, definition, name, cb) {
        try {
            var model = definition(connection, connection.base.Schema, name);
            return cb(null, model);
        } catch (e) {
            return cb(e);
        }
    }
}
```

**Comment by bschnelle** [comment]
where is the `name` argument in the `registerModel` function coming from?  continuing to use `definition.__name` from the file name?

**Comment by cludden** [comment]
no, in the comment above, it would be if we added name as the third argument passed to the registerModel hook. currently it would come from the definition.

**Comment by cludden** [comment]
do you see an issue with adding `name` as a third and optional argument to pass to the `registerModel` adapter hook? that's kind of the way i'm leaning right now

**Comment by bschnelle** [comment]
I must be missing something.  `adapter.registerModel` is called by the `models` hook so if we add the optional `name` argument then how is that getting from the model definition to the `registerModel` function?  In your example above I don't see where `name` is being defined prior to `registerModel` being called.

File: test/test-adapters/two.js [hunk 1]
```diff
@@ -61,5 +61,18 @@ module.exports = {
...
+    },
+
+    blueprintController: function(model, actions, cb) {
+        var controller = {};
+
+        actions.forEach(function(action) {
+            controller[action] = function(req, res, next) {
+                res.send(200, 'blueprint ' + action);
+                next();
+            }
```
Comment (cludden) [inline test/test-adapters/two.js:73]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573835/issues/source?bid=3045492&fileBranchId=3056295#l73)

File: test/hooks/blueprints.test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/* jshint expr:true */
+'use strict';
+
+var expect = require('chai').expect,
+    Microservice = require('../../index'),
+    supertest = require('supertest');
```
Comment (cludden) [inline test/hooks/blueprints.test.js:6]: Codacy Issue found: ["supertest" is defined but never used](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573794/issues/source?bid=3045492&fileBranchId=3056295#l6)

File: test/hooks/blueprints.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+var cwd,
+    request,
```
Comment (cludden) [inline test/hooks/blueprints.test.js:10]: Codacy Issue found: ["request" is defined but never used](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573794/issues/source?bid=3045492&fileBranchId=3056295#l10)

File: hooks/models.js [hunk 4]
```diff
@@ -32,13 +32,17 @@ module.exports = function Models(cb) {
...
                 // get model definition
                 var modelDefinition = modelDefinitions[name];
-                modelDefinition.__name = name;
 
                 // get connection info
                 var connectionInfo = lib.findConnection(microservice, name);
                 if (!connectionInfo) {
                     microservice.log('silly', '[models] Unable to find explicit adapter for model (' + name + ')');
                     if (!defaultConnection ) {
+                        console.log('*******************');
```
Comment (cludden) [inline hooks/models.js:41]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573785/issues/source?bid=3045492&fileBranchId=3056295#l41)

File: hooks/models.js [hunk 5]
```diff
@@ -32,13 +32,17 @@ module.exports = function Models(cb) {
...
+                        console.log('*******************');
+                        console.log(microservice.connections);
```
Comment (cludden) [inline hooks/models.js:43]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573785/issues/source?bid=3045492&fileBranchId=3056295#l43)

File: test/hooks/routes.test.js [hunk 6]
```diff
@@ -20,10 +20,11 @@ describe('[hook] routes', function() {
         process.chdir(__dirname + '/routes/test-app-no-problems');
 
         var _microservice = new Microservice({});
-
+        console.log('STARTING ROUTE TEST ******************************************')
```
Comment (cludden) [inline test/hooks/routes.test.js:23]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573810/issues/source?bid=3045492&fileBranchId=3056295#l23)

File: test/hooks/routes.test.js [hunk 7]
```diff
@@ -20,10 +20,11 @@ describe('[hook] routes', function() {
...
         _microservice.start(function(err) {
             process.chdir(cwd);
             expect(err).to.not.exist;
+            console.log('ENDING ROUTE TEST ******************************************')
```
Comment (cludden) [inline test/hooks/routes.test.js:27]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573810/issues/source?bid=3045492&fileBranchId=3056295#l27)

Comment (cludden) [inline test/hooks/routes.test.js:27]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573810/issues/source?bid=3045492&fileBranchId=3056295#l27)

File: test/hooks/routes.test.js (hunk 6 above)
Comment (cludden) [inline test/hooks/routes.test.js:23]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573810/issues/source?bid=3045492&fileBranchId=3056295#l23)

File: index.js [hunk 8]
```diff
@@ -88,6 +88,8 @@ var Microservice = function(config) {
     });
 
     self.start = function(done) {
+        console.log('CONFIG', self._config.server);
```
Comment (cludden) [inline index.js:91]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/chris-ludden/restify-microservice/file/1745573792/issues/source?bid=3045492&fileBranchId=3056295#l91)

**Pull request closed without merging**